File: src/test/java/com/service/StatisticsComputationEmptyListTest.java
package com.service;

import com.model.Statistics;
import com.service.serviceImpl.StatisticsComputationImpl;
import org.junit.Before;
import org.junit.Test;

import java.math.BigDecimal;
import java.util.ArrayList;

import static org.junit.Assert.assertEquals;

public class StatisticsComputationEmptyListTest {

    private StatisticsComputation statisticsComputation;
    private ArrayList<BigDecimal> emptyList;


    @Before
    public void setData(){
        emptyList = new ArrayList<>();
        statisticsComputation = new StatisticsComputationImpl();
    }

    @Test
    public void computeSumRoundUp(){
        Statistics statistics = statisticsComputation.computeStatistics(emptyList);
        assertEquals(BigDecimal.valueOf(000,2),statistics.getSum());
    }

    @Test
    public void computeAverageRoundUp(){
        Statistics statistics = statisticsComputation.computeStatistics(emptyList);
        assertEquals(BigDecimal.valueOf(000,2),statistics.getAvg());
    }

    @Test
    public void computeMaxRoundUp(){
        Statistics statistics = statisticsComputation.computeStatistics(emptyList);
        assertEquals(BigDecimal.valueOf(000,2),statistics.getMax());
    }

    @Test
    public void computeMinRoundUp(){
        Statistics statistics = statisticsComputation.computeStatistics(emptyList);
        assertEquals(BigDecimal.valueOf(000,2),statistics.getMin());
    }
}

File: src/test/kotlin/com/service/StatisticsComputationEmptyListTest.kt
package com.service

import java.math.BigDecimal
import com.service.serviceImpl.StatisticsComputationImpl
import org.junit.Assert
import org.junit.Test


class StatisticsComputationEmptyListTest {
    private var statisticsComputation: StatisticsComputation = StatisticsComputationImpl()
    private var emptyList: List<BigDecimal> = emptyList()

    @Test
    fun computeSumRoundUp() {
        val sum = statisticsComputation.computeStatistics(emptyList).sum
        Assert.assertEquals(BigDecimal.valueOf(0, 2), sum)
    }

    @Test
    fun computeAverageRoundUp() {
        val avg = statisticsComputation.computeStatistics(emptyList).avg
        Assert.assertEquals(BigDecimal.valueOf(0, 2), avg)
    }

    @Test
    fun computeMaxRoundUp() {
        val max = statisticsComputation.computeStatistics(emptyList).max
        Assert.assertEquals(BigDecimal.valueOf(0, 2), max)
    }

    @Test
    fun computeMinRoundUp() {
        val min = statisticsComputation.computeStatistics(emptyList).min
        Assert.assertEquals(BigDecimal.valueOf(0, 2), min)
    }
}